docs/proposals/PROPOSAL_B.md

```diff
@@ -0,0 +1,287 @@
+A plugin is an executable binary that validates the host and returns information about success or failure.
+
+### Compatibility Spec
+
+Compatibility spec is a graph that can have multiple roots with different paths.
+If one namespace in the path fails, the path is incompatible with the host.
+Namespace fails if at least one plugin reports an error.
+If all paths fail, the host is not compatible with the container.
+
+The host is compatible with the container if at least one path is completed successfully.
+
+#### Schema
+
+- **`schemaVersion`** *int*
+
+  This REQUIRED property specifies the image manifest schema version. For this version of the specification.
+
+- **`namespaces`** *string-object map*
+
+  This REQUIRED property specifies a map of validation areas.
+
+  - **`<namespace-name>`** *object*
+
+    This REQUIRED property specifies the validation area.
+
+    - **`plugins`** *string-object*
+
+      This OPTIONAL property is a list of plugins specifications.
+
+        - **`<plugin-name>`** *object*
+
+          This REQUIRED property specifies the plugin specification for plugin executable binary that validates the host and returns information about success or failure
+
+          - **`source`** *string*
+
+            This OPTIONAL property specifies an HTTP address of the plugin.
+
+          - **`data`** *object*
+
+            This OPTIONAL property specifies the input data for plugin executable binary. 
+
+    - **`refs`** *object*
+
+      This OPTIONAL property specifies the references to other namespaces.
+
+      - **`all`** *array of strings*
+
+        This OPTIONAL property specifies a list of all required namespaces that have to pass that container is considered to be compatible with a host.
+
+      - **`oneOf`** *array of strings*
+
+        This OPTIONAL property specifies a list of namespaces with a condition that at least one namespace has to pass that container is considered to be compatible with a host.
+
+      - **`noneOf`** *array of strings* 
+
+        This OPTIONAL property specifies a list of namespaces that cannot pass that container is considered to be compatible with a host.
+
+##### Example
+
+<details>
+  <summary>GPU pass-through</summary>
+
+```json
+{
+  "schema": "0.1.0",
+  "namespaces": {
+    "kernelMainlineV5": {
+      "plugins": {
+        "kernelVersion": {
+          "source": "<http-something-to-download-plugin>",
+          "data": {
+            "range": "5.0.0-5.15.0"
+          }
+        },
+        "refs": {
+          "all": [
+            "vfio",
+            "cpu",
+            "gpu"
+          ]
+        }
+      }
+    },
+    "vfio": {
+      "plugins": {
+        "kernelConfiguration": {
+          "data": {
+            "CONFIG_VFIO": "m",
+            "CONFIG_VFIO_IOMMU_TYPE1": "m",
+            "CONFIG_VFIO_MDEV": "m",
+            "CONFIG_VFIO_MDEV_DEVICE": "m",
+            "CONFIG_VFIO_PCI": "m"
```

Comment:
As long as the kernel config files have the same defines, but these are not stable over time. Should this be a "meta" VFIO tag/capability that would be verified by either sysadmin or a script/containerized test? "y" is also acceptable, thus value should be enabled/disabled/1/0/y/n as expectation is that kernel modules do get loaded?

Reply:
Your comment is valid but I think we could leave this discussion for the kernelConfiguration plugin itself. This is just an example of how plugins could be used with spec. I will keep the comment for the future.